publicip/src/lib.rs

```diff
@@ -137,24 +133,19 @@ impl<T: Into<IpAddr> + FromStr<Err = AddrParseError>> Request<T> {
         if qname_len & 0xc000 != 0xc000 {
             buf.set_position(buf.position() + qname_len as u64);
         }
-        ensure!(buf.read_u16()? == TYPE_TXT, "answer is not TXT type");
-        ensure!(buf.read_u16()? == CLASS_CH, "answer is not CH class");
+        ensure!(
+            buf.read_u16()? == self.record_type,
+            "answer is not expected type"
+        );
```

Comment:
Would be nice to include the expected type in the error message like eg.

```suggestion
        let answer_type = buf.read_u16()?;
        ensure!(
            answer_type == self.record_type,
            "answer type ({answer_type}) is not expected type ({self.record_type})"
        );
```

but I see that would require touching up our `ensure!` macro which is probably out of scope here.

Reply:
yep that was my rationale for the ambiguous message too :P.